lib/child_process.js

```diff
@@ -437,6 +437,11 @@ function execFile(file, args = [], options, callback) {
       child.stdout.setEncoding(encoding);
 
     child.stdout.on('data', function onChildStdout(chunk) {
+      // Do not need to count the length
+      if (options.maxBuffer === Infinity) {
```

Comment:
I have no objections but I think it does not make much sense to create a fast path for a value that is not common and the not the default value.

Reply:
Yes, but we would benefit from it when set the `maxBuffer` to infinity, so I think it's worth it. and the fast path is easy to implement.